### PR TITLE
Set object level permissions to update/delete a challenge.

### PR DESCRIPTION
**Pull request by RishabhJain2018**
#174 solved. @deshraj @TroJan Please review the PR.

### CODE REVIEW

File: apps/challenges/permissions.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+from rest_framework import permissions
+from .serializers import ChallengeSerializer
+from .models import Challenge
+
+
+class ChallengePermission(permissions.BasePermission):
```
Comment (deshraj) [inline apps/challenges/permissions.py]: Lets change the Class name to `IsChallengeCreator`

File: apps/challenges/permissions.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+from rest_framework import permissions
+from .serializers import ChallengeSerializer
```
Comment (TroJan) [inline apps/challenges/permissions.py]: can you please add a new line here?

File: apps/challenges/permissions.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+from .models import Challenge
```
Comment (TroJan) [inline apps/challenges/permissions.py]: also can you please reorder your imports like

```
from .models import .....
from .serializers import ...
```

File: apps/challenges/permissions.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+class ChallengePermission(permissions.BasePermission):
+    """
+    Permission class for edit/delete a challenge.
+    """
+
+    def has_permission(self, request, view):
+
+        if request.method in permissions.SAFE_METHODS:
+            return True
+        elif request.method == "PUT" or request.method == "PATCH" or request.method == "DELETE":
```
Comment (TroJan) [inline apps/challenges/permissions.py]: also can you please rewrite the elif condition to use `in` like

```
elif request.method in ['DELETE', 'PATCH', 'PUT']:
```

File: apps/challenges/views.py [hunk 5]
```diff
@@ -14,6 +14,7 @@
 
 from .models import Challenge
 from .serializers import ChallengeSerializer
+from .permissions import ChallengePermission
```
Comment (TroJan) [inline apps/challenges/views.py]: can you please reorder your import alphabetically ?

File: apps/challenges/permissions.py (hunk 1 above)
Comment (RishabhJain2018) [inline apps/challenges/permissions.py]: yeah sure.

File: apps/challenges/permissions.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+            challenge = Challenge.objects.filter(pk=request.parser_context['kwargs']['pk'])
```
Comment (RishabhJain2018) [inline apps/challenges/permissions.py]: okk.

File: apps/challenges/permissions.py (hunk 4 above)
Comment (RishabhJain2018) [inline apps/challenges/permissions.py]: yeah sure.

File: apps/challenges/views.py (hunk 5 above)
Comment (RishabhJain2018) [inline apps/challenges/views.py]: okk sure.

File: apps/challenges/permissions.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
+from rest_framework import permissions
+from .models import Challenge
```
Comment (deshraj) [inline apps/challenges/permissions.py]: @RishabhJain2018 there should be a gap between line 1 and line 2. Also, there should be no gap between the lines `from .models import Challenge` and `from .serializers import ChallengeSerializer`

**Comment by deshraj** [comment]
Looks good. :+1:

**Comment by RishabhJain2018** [comment]
@deshraj @TroJan Please review the PR. I have solved the merge conflicts.

**Comment by TroJan** [comment]
looks okay to me.

**Pull request merged**